node/node_genesis.go

```diff
@@ -47,14 +50,12 @@ func (gi *genesisInitializer) InitChainDB(db ethdb.Database, shardID uint32) err
 		}
 		shardState = types.ShardState{*c}
 	}
-	if err := gi.node.SetupGenesisBlock(db, shardID, shardState); err != nil {
-		return ctxerror.New("cannot setup genesis block").WithCause(err)
-	}
+	gi.node.SetupGenesisBlock(db, shardID, shardState)
 	return nil
 }
 
 // SetupGenesisBlock sets up a genesis blockchain.
-func (node *Node) SetupGenesisBlock(db ethdb.Database, shardID uint32, myShardState types.ShardState) error {
+func (node *Node) SetupGenesisBlock(db ethdb.Database, shardID uint32, myShardState types.ShardState) {
```

Comment:
why error is ignored?

Reply:
Switched to panic for bad genesis block.